src/usr/share/netcheck/netcheck.py

```diff
@@ -776,3 +794,55 @@ def _calculate_next_log_time(self, loop_time):
         Returns the datetime that the logging should occur.
         """
         return loop_time + datetime.timedelta(seconds=self.config['periodic_status_delay'])
+
+    def _check_for_gateway_change(self):
+        """Checks the current state of the default gateway and issues a broadcast if it is
+        different from the prior state.
+        """
+
+        new_gateway_state = self._get_default_gateway_state()
+
+        if new_gateway_state != self.default_gateway_state:
+            self.logger.info(
+                'The default gateway has changed to %s via %s on interface %s.',
+                new_gateway_state['address'],
+                new_gateway_state['connection_id'],
+                new_gateway_state['interface'])
+
+            self.default_gateway_state = new_gateway_state
+            self.broadcaster.issue()
+
+    def _get_default_gateway_state(self):
+        """Retrieves information about the current default gateway.
+
+        Returns a dictionary containing a gateway IP address, interface name, and associated
+        connection id.
```

Comment:
I'd say "Returns a dictionary containing the gateway IP address, interface name, and associated connection ID."

Reply:
Also, the second line of the return sentence should be reverse indented.

Reply:
Also, specify what happens if there is no primary default gateway.